src/creds/VerifiableCredential.js

```diff
@@ -769,17 +788,27 @@ VerifiableCredentialBaseConstructor.fromJSON = (verifiableCredentialJSON) => {
 /**
  * List all properties of a Verifiable Credential
  */
-VerifiableCredentialBaseConstructor.getAllProperties = (identifier) => {
+VerifiableCredentialBaseConstructor.getAllProperties = async (identifier) => {
+  await schemaLoader.loadSchemaFromTitle(identifier);
+
   const vcDefinition = _.find(definitions, { identifier });
   if (vcDefinition) {
     const allProperties = [];
-    _.forEach(vcDefinition.depends, (ucaIdentifier) => {
-      allProperties.push(...Claim.getAllProperties(ucaIdentifier));
-    });
+    // eslint-disable-next-line no-restricted-syntax
+    for (const definition of vcDefinition.depends) {
+      // eslint-disable-next-line no-await-in-loop
+      allProperties.push(...await Claim.getAllProperties(definition));
```

Comment:
await + spread this is really bad, it is poor readability
stop disabling the lint btw, change the syntax